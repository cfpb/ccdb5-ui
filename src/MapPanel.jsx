import './MapPanel.less'
import ActionBar from './ActionBar'
import { connect } from 'react-redux'
import DateIntervals from './DateIntervals'
import Loading from './Dialogs/Loading'
import React from 'react'
import RowChart from './RowChart'
import TileChartMap from './TileChartMap'

export class MapPanel extends React.Component {
  render() {
    return (
      <section className="map-panel">
        <ActionBar/>
        <DateIntervals/>
        <TileChartMap/>
        <RowChart aggtype="product" />
        <RowChart aggtype="issue" />
        <Loading isLoading={ this.props.isLoading || false }/>
      </section>
    )
  }
}

const mapStateToProps = state => ( {
  isLoading: state.results.isLoading,
  items: state.results.items
} )

export default connect( mapStateToProps )( MapPanel )
