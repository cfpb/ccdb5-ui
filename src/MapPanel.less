@import (less) "base.less";

.map-panel {
  .cards-panel {
    padding: @grid_gutter-width;
    margin: 0;
 
   @media @phone {
    padding: @gutter-normal;
  }
 }

  h2 {
    padding: @gutter-normal;    
  }
}