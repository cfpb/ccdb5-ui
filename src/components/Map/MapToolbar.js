import './MapToolbar.scss';
import { useDispatch, useSelector } from 'react-redux';
import getIcon from '../Common/Icon/iconMap';
import { MODE_LIST, THESE_UNITED_STATES } from '../../constants';
import { stateFilterCleared } from '../../reducers/filters/filtersSlice';
import { selectFiltersState } from '../../reducers/filters/selectors';
import { tabChanged } from '../../reducers/view/viewSlice';

export const MapToolbar = () => {
  const dispatch = useDispatch();
  const stateFilters = useSelector(selectFiltersState);
  const filteredStates = stateFilters
    ? stateFilters
        .filter((state) => state in THESE_UNITED_STATES)
        .map((state) => THESE_UNITED_STATES[state])
        .join(', ')
    : '';

  return (
    <div className="map-toolbar">
      <section className="state-heading">
        {!filteredStates && <span>United States of America</span>}
        <span>{filteredStates}</span>
        {!!filteredStates && (
          <button
            aria-label="Clear all map filters"
            className="a-btn a-btn--link"
            onClick={() => {
              dispatch(stateFilterCleared());
            }}
          >
            {getIcon('delete-round')}
            Clear
          </button>
        )}
      </section>
      {!!filteredStates && (
        <section className="state-navigation">
          <button
            className="list a-btn a-btn--link"
            onClick={() => {
              dispatch(tabChanged(MODE_LIST));
            }}
          >
            View complaints for filtered states
          </button>
        </section>
      )}
    </div>
  );
};
