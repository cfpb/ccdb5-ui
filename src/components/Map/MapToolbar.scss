.map-toolbar {
  border-top: 2px solid var(--gray-20);
  border-bottom: 1px solid var(--gray-20);
  margin: 0 20px;
  a {
    text-decoration: underline;
    &:link,
    &:hover,
    &:active,
    &:visited {
      cursor: pointer;
    }
  }
  .state {
    &-heading {
      padding: 0.625rem;
      border-bottom: 1px solid var(--gray-20);
      span {
        margin: 0.625rem;
      }
      svg {
        padding-right: 0.313rem;
      }
    }
    &-navigation {
      padding: 0.625rem;
    }
  }
}
