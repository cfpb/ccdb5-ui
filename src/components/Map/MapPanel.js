import '../RefineBar/RefineBar.less';
import { ActionBar } from '../ActionBar/ActionBar';
import { useDispatch, useSelector } from 'react-redux';
import ErrorBlock from '../Warnings/Error';
import { FilterPanel } from '../Filters/FilterPanel';
import { FilterPanelToggle } from '../Filters/FilterPanelToggle';
import { Loading } from '../Loading/Loading';
import { MapToolbar } from './MapToolbar';
import { mapWarningDismissed } from '../../reducers/filters/filtersSlice';
import { PerCapita } from '../RefineBar/PerCapita';
import { processRows } from '../../utils/chart';

import { useMemo } from 'react';
import { RowChart } from '../Charts/RowChart/RowChart';
import { Separator } from '../RefineBar/Separator';
import { TabbedNavigation } from '../TabbedNavigation';
import { TileChartMap } from './TileChartMap/TileChartMap';
import Warning from '../Warnings/Warning';
import { selectAggsTotal } from '../../reducers/aggs/selectors';

import {
  selectFiltersEnablePer1000,
  selectFiltersMapWarningEnabled,
} from '../../reducers/filters/selectors';
import {
  selectMapActiveCall,
  selectMapError,
  selectMapResults,
} from '../../reducers/map/selectors';

import {
  selectQueryDateReceivedMax,
  selectQueryDateReceivedMin,
} from '../../reducers/query/selectors';

import {
  selectViewExpandedRows,
  selectViewWidth,
} from '../../reducers/view/selectors';

import { shortFormat } from '../../utils';

const WARNING_MESSAGE =
  '“Complaints per 1,000 population” is not available with your filter ' +
  'selections.';

const MAP_ROWCHART_HELPERTEXT =
  'Product the consumer identified in the complaint. Click on a product ' +
  'to expand sub-products';

export const MapPanel = () => {
  const dispatch = useDispatch();
  const total = useSelector(selectAggsTotal);

  const enablePer1000 = useSelector(selectFiltersEnablePer1000);
  const mapWarningEnabled = useSelector(selectFiltersMapWarningEnabled);

  const activeCall = useSelector(selectMapActiveCall);
  const results = useSelector(selectMapResults);
  const hasError = useSelector(selectMapError);

  const maxDate = useSelector(selectQueryDateReceivedMax);
  const minDate = useSelector(selectQueryDateReceivedMin);

  const expandedRows = useSelector(selectViewExpandedRows);
  const width = useSelector(selectViewWidth);
  const hasMobileFilters = width < 750;
  const hasWarning = !enablePer1000 && mapWarningEnabled;
  const productData = useMemo(() => {
    return processRows(results.product, false, 'Product', expandedRows);
  }, [results, expandedRows]);

  const MAP_ROWCHART_TITLE = `Product by highest complaint volume ${shortFormat(
    minDate,
  )} to ${shortFormat(maxDate)}`;

  const onDismissWarning = () => {
    dispatch(mapWarningDismissed());
  };

  return (
    <section className="map-panel">
      <ActionBar />
      <TabbedNavigation />
      {!!hasError && (
        <ErrorBlock text="There was a problem executing your search" />
      )}
      {!!hasWarning && (
        <Warning text={WARNING_MESSAGE} closeFn={onDismissWarning} />
      )}
      {!!hasMobileFilters && <FilterPanel />}
      <FilterPanelToggle />
      <div className="layout-row refine-bar">
        <Separator />
        <PerCapita />
      </div>
      <TileChartMap />
      <MapToolbar />
      <RowChart
        id="product"
        colorScheme={productData.colorScheme}
        data={productData.data}
        title={MAP_ROWCHART_TITLE}
        helperText={MAP_ROWCHART_HELPERTEXT}
        total={total}
      />

      <Loading isLoading={activeCall !== ''} />
    </section>
  );
};
