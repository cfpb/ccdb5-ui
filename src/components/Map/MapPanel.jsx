import '../RefineBar/RefineBar.less'
import ActionBar from '../ActionBar'
import { connect } from 'react-redux'
import DateRanges from '../RefineBar/DateRanges'
import ErrorBlock from '../Warnings/Error'
import FilterPanel from '../Filters/FilterPanel'
import FilterPanelToggle from '../Filters/FilterPanelToggle'
import Loading from '../Dialogs/Loading'
import MapToolbar from './MapToolbar'
import { mapWarningDismissed } from '../../actions/view'
import { PerCapita } from '../RefineBar/PerCapita'
import { processRows } from '../../utils/chart'
import React from 'react'
import RowChart from '../Charts/RowChart'
import { Separator } from '../RefineBar/Separator'
import { shortFormat } from '../../utils'
import TabbedNavigation from '../TabbedNavigation'
import TileChartMap from '../Charts/TileChartMap'
import Warning from '../Warnings/Warning'


const WARNING_MESSAGE = '“Complaints per' +
  ' 1,000 population” is not available with your filter selections.'

const MAP_ROWCHART_HELPERTEXT = 'Product the consumer identified in the' +
' complaint. Click on a product to expand sub-products'

export class MapPanel extends React.Component {
  // eslint-disable-next-line complexity
  render() {
    const MAP_ROWCHART_TITLE = 'Product by highest complaint volume' +
      ' ' + this.props.minDate + ' to ' + this.props.maxDate

    return (
      <section className="map-panel">
        <ActionBar/>
        <TabbedNavigation />
        { this.props.error &&
          <ErrorBlock text="There was a problem executing your search" />
        }
        { this.props.showWarning &&
          <Warning text={ WARNING_MESSAGE }
                   closeFn={this.props.onDismissWarning}/> }
        { this.props.showMobileFilters && <FilterPanel/> }
        <div className="layout-row refine-bar">
          <FilterPanelToggle/>
          <Separator />
          <DateRanges/>
          <PerCapita/>
        </div>
        <TileChartMap/>
        <MapToolbar/>
        <RowChart id="product"
                  colorScheme={this.props.productData.colorScheme}
                  data={this.props.productData.data}
                  title={ MAP_ROWCHART_TITLE }
                  helperText={ MAP_ROWCHART_HELPERTEXT }
                  total={ this.props.total }/>

        <Loading isLoading={ this.props.isLoading || false }/>
      </section>
    )
  }
}

const mapStateToProps = state => {
  const {
    error,
    isLoading,
    results
  } = state.map

  const {
    date_received_max: maxDate,
    date_received_min: minDate,
    enablePer1000,
    mapWarningEnabled
  } = state.query

  const { expandedRows, width } = state.view

  return {
    error,
    isLoading,
    minDate: shortFormat( minDate ),
    maxDate: shortFormat( maxDate ),
    productData: processRows( results.product, false, 'Product', expandedRows ),
    showMobileFilters: width < 750,
    showWarning: !enablePer1000 && mapWarningEnabled,
    total: state.aggs.total
  }
}

export const mapDispatchToProps = dispatch => ( {
  onDismissWarning: () => {
    dispatch( mapWarningDismissed() )
  }
} )


export default connect( mapStateToProps, mapDispatchToProps )( MapPanel )
