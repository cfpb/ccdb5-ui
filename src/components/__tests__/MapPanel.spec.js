import { mapDispatchToProps } from '../Map/MapPanel';

describe('component:MapPanel', () => {
  describe('mapDispatchToProps', () => {
    it('hooks into dismissWarning', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).onDismissWarning();
      expect(dispatch.mock.calls.length).toEqual(1);
    });
  });
});
