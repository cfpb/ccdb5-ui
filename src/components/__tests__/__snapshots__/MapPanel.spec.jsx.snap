// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:MapPanel renders Print without crashing 1`] = `
<section
  className="map-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            2
          </span>
           matches out of 
          <span>
            100
          </span>
           total complaints
        </h2>
      </div>
      <div>
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link export-btn"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="tabbed-navigation"
  >
    <section>
      <button
        className="trends tab"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 15 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.55 15.256H1.45a.554.554 0 0 1-.553-.554V3.168a.554.554 0 1 1 1.108 0v10.98h11.544a.554.554 0 0 1 0 1.108zM3.121 13.02V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.785 0V3.507a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v9.513zm2.785 0V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.786 0v-2.753a.476.476 0 0 1 .475-.475h.785a.476.476 0 0 1 .475.475v2.753z"
          />
        </svg>
        Trends
      </button>
      <button
        className="list tab"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 14 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.85 4.836a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm1.13-8.692a1.03 1.03 0 0 1 1.029-1.029h7.173a1.03 1.03 0 1 1 0 2.058H5.009a1.03 1.03 0 0 1-1.03-1.029zm9.231 4.347a1.03 1.03 0 0 1-1.029 1.029H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03zm0 4.345a1.03 1.03 0 0 1-1.029 1.03H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03z"
          />
        </svg>
        List
      </button>
      <button
        className="map tab active"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 16 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.084 15.2H.916a.264.264 0 0 1-.254-.42l2.36-4.492a.865.865 0 0 1 .696-.42h.827a9.51 9.51 0 0 0 .943 1.108H3.912l-1.637 3.116h11.45l-1.637-3.116h-1.34a9.481 9.481 0 0 0 .943-1.109h.591a.866.866 0 0 1 .696.421l2.36 4.492a.264.264 0 0 1-.254.42zM11.4 7.189c0 2.64-2.176 2.888-3.103 5.46a.182.182 0 0 1-.356 0c-.928-2.572-3.104-2.82-3.104-5.46a3.282 3.282 0 0 1 6.563 0zm-1.86-.005a1.425 1.425 0 1 0-1.425 1.425A1.425 1.425 0 0 0 9.54 7.184z"
          />
        </svg>
        Map
      </button>
    </section>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
        >
          Filter results
        </button>
      </div>
    </section>
    <span
      className="separator"
    />
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <section
      className="m-btn-group"
    >
      <p>
        Map shading
      </p>
      <button
        className="a-btn toggle-button raw selected"
        onClick={[Function]}
      >
        Complaints
      </button>
      <button
        className="a-btn toggle-button capita deselected"
        onClick={[Function]}
      >
        Complaints per 1,000 
        <span>
          population
        </span>
      </button>
    </section>
  </div>
  <div>
    <div
      className="cfpb-chart print"
      data-chart-type="tile_map"
      id="tile-chart-map"
    />
  </div>
  <div
    className="map-toolbar"
  >
    <section
      className="state-heading"
    >
      <span>
        United States of America
      </span>
      <span>
        
      </span>
      
    </section>
    
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume 7/10/2017 to 7/10/2020
    </h3>
    <p>
      Product the consumer identified in the complaint. Click on a product to expand sub-products
    </p>
    <div
      id="row-chart-product"
    />
  </div>
</section>
`;

exports[`component:MapPanel renders warning without crashing 1`] = `
<section
  className="map-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            2
          </span>
           matches out of 
          <span>
            100
          </span>
           total complaints
        </h2>
      </div>
      <div>
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link export-btn"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="tabbed-navigation"
  >
    <section>
      <button
        className="trends tab"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 15 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.55 15.256H1.45a.554.554 0 0 1-.553-.554V3.168a.554.554 0 1 1 1.108 0v10.98h11.544a.554.554 0 0 1 0 1.108zM3.121 13.02V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.785 0V3.507a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v9.513zm2.785 0V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.786 0v-2.753a.476.476 0 0 1 .475-.475h.785a.476.476 0 0 1 .475.475v2.753z"
          />
        </svg>
        Trends
      </button>
      <button
        className="list tab"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 14 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.85 4.836a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm1.13-8.692a1.03 1.03 0 0 1 1.029-1.029h7.173a1.03 1.03 0 1 1 0 2.058H5.009a1.03 1.03 0 0 1-1.03-1.029zm9.231 4.347a1.03 1.03 0 0 1-1.029 1.029H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03zm0 4.345a1.03 1.03 0 0 1-1.029 1.03H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03z"
          />
        </svg>
        List
      </button>
      <button
        className="map tab active"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 16 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.084 15.2H.916a.264.264 0 0 1-.254-.42l2.36-4.492a.865.865 0 0 1 .696-.42h.827a9.51 9.51 0 0 0 .943 1.108H3.912l-1.637 3.116h11.45l-1.637-3.116h-1.34a9.481 9.481 0 0 0 .943-1.109h.591a.866.866 0 0 1 .696.421l2.36 4.492a.264.264 0 0 1-.254.42zM11.4 7.189c0 2.64-2.176 2.888-3.103 5.46a.182.182 0 0 1-.356 0c-.928-2.572-3.104-2.82-3.104-5.46a3.282 3.282 0 0 1 6.563 0zm-1.86-.005a1.425 1.425 0 1 0-1.425 1.425A1.425 1.425 0 0 0 9.54 7.184z"
          />
        </svg>
        Map
      </button>
    </section>
  </div>
  <div
    className="warning m-notification m-notification__visible m-notification__warning"
  >
    <svg
      className="cf-icon-svg"
      viewBox="0 0 1000 1200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm-49.7 234.6c0-27.6 22.4-50 50-50s50 22.4 50 50v328.6c0 27.6-22.4 50-50 50s-50-22.4-50-50V339.8zm50 582.5c-39.6 0-71.7-32.1-71.7-71.7s32.1-71.7 71.7-71.7S572 811 572 850.6s-32.1 71.7-71.7 71.7z"
      />
    </svg>
    <div
      className="m-notification_content"
    >
      <div
        className="h4 m-notification_message"
      >
        “Complaints per 1,000 population” is not available with your filter selections.
      </div>
    </div>
    <span
      aria-label="Dismiss"
      className="close"
      onClick={[Function]}
    >
      <svg
        className="cf-icon-svg"
        viewBox="0 0 718.9 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
        />
      </svg>
    </span>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
        >
          Filter results
        </button>
      </div>
    </section>
    <span
      className="separator"
    />
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <section
      className="m-btn-group"
    >
      <p>
        Map shading
      </p>
      <button
        className="a-btn toggle-button raw selected"
        onClick={[Function]}
      >
        Complaints
      </button>
      <button
        className="a-btn toggle-button capita a-btn__disabled"
        onClick={[Function]}
      >
        Complaints per 1,000 
        <span>
          population
        </span>
      </button>
    </section>
  </div>
  <div>
    <div
      className="cfpb-chart "
      data-chart-type="tile_map"
      id="tile-chart-map"
    />
  </div>
  <div
    className="map-toolbar"
  >
    <section
      className="state-heading"
    >
      <span>
        United States of America
      </span>
      <span>
        
      </span>
      
    </section>
    
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume 7/10/2017 to 7/10/2020
    </h3>
    <p>
      Product the consumer identified in the complaint. Click on a product to expand sub-products
    </p>
    <div
      id="row-chart-product"
    />
  </div>
</section>
`;

exports[`component:MapPanel renders without crashing 1`] = `
<section
  className="map-panel"
>
  <div>
    <summary
      className="action-bar"
      id="search-summary"
    >
      <div>
        <h2>
          Showing 
          <span>
            2
          </span>
           matches out of 
          <span>
            100
          </span>
           total complaints
        </h2>
      </div>
      <div>
        <h3
          className="h4 flex-all export-results"
        >
          <button
            className="a-btn a-btn__link export-btn"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export data
          </button>
          <button
            className="a-btn a-btn__link print-preview"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 761.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
              />
              <path
                d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
              />
            </svg>
            Print
          </button>
        </h3>
      </div>
    </summary>
  </div>
  <div
    className="tabbed-navigation"
  >
    <section>
      <button
        className="trends tab"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 15 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.55 15.256H1.45a.554.554 0 0 1-.553-.554V3.168a.554.554 0 1 1 1.108 0v10.98h11.544a.554.554 0 0 1 0 1.108zM3.121 13.02V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.785 0V3.507a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v9.513zm2.785 0V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.786 0v-2.753a.476.476 0 0 1 .475-.475h.785a.476.476 0 0 1 .475.475v2.753z"
          />
        </svg>
        Trends
      </button>
      <button
        className="list tab"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 14 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.85 4.836a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm1.13-8.692a1.03 1.03 0 0 1 1.029-1.029h7.173a1.03 1.03 0 1 1 0 2.058H5.009a1.03 1.03 0 0 1-1.03-1.029zm9.231 4.347a1.03 1.03 0 0 1-1.029 1.029H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03zm0 4.345a1.03 1.03 0 0 1-1.029 1.03H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03z"
          />
        </svg>
        List
      </button>
      <button
        className="map tab active"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 16 19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.084 15.2H.916a.264.264 0 0 1-.254-.42l2.36-4.492a.865.865 0 0 1 .696-.42h.827a9.51 9.51 0 0 0 .943 1.108H3.912l-1.637 3.116h11.45l-1.637-3.116h-1.34a9.481 9.481 0 0 0 .943-1.109h.591a.866.866 0 0 1 .696.421l2.36 4.492a.264.264 0 0 1-.254.42zM11.4 7.189c0 2.64-2.176 2.888-3.103 5.46a.182.182 0 0 1-.356 0c-.928-2.572-3.104-2.82-3.104-5.46a3.282 3.282 0 0 1 6.563 0zm-1.86-.005a1.425 1.425 0 1 0-1.425 1.425A1.425 1.425 0 0 0 9.54 7.184z"
          />
        </svg>
        Map
      </button>
    </section>
  </div>
  <div
    className="layout-row refine-bar"
  >
    <section
      className="filter-panel-toggle"
    >
      <div
        className="m-btn-group"
      >
        <p>
           
        </p>
        <button
          className="a-btn"
          onClick={[Function]}
        >
          Filter results
        </button>
      </div>
    </section>
    <span
      className="separator"
    />
    <section
      className="date-ranges m-btn-group"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="a-btn date-selector range-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="a-btn date-selector range-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="a-btn date-selector range-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="a-btn date-selector range-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="a-btn date-selector range-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <section
      className="m-btn-group"
    >
      <p>
        Map shading
      </p>
      <button
        className="a-btn toggle-button raw selected"
        onClick={[Function]}
      >
        Complaints
      </button>
      <button
        className="a-btn toggle-button capita deselected"
        onClick={[Function]}
      >
        Complaints per 1,000 
        <span>
          population
        </span>
      </button>
    </section>
  </div>
  <div>
    <div
      className="cfpb-chart "
      data-chart-type="tile_map"
      id="tile-chart-map"
    />
  </div>
  <div
    className="map-toolbar"
  >
    <section
      className="state-heading"
    >
      <span>
        United States of America
      </span>
      <span>
        
      </span>
      
    </section>
    
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product by highest complaint volume 7/10/2017 to 7/10/2020
    </h3>
    <p>
      Product the consumer identified in the complaint. Click on a product to expand sub-products
    </p>
    <div
      id="row-chart-product"
    />
  </div>
</section>
`;
